### ISSUE TITLE
poor performance with react-native 0.53

### ISSUE THREAD

**Issue by macrozone** [jhen0409/react-native-debugger#197]
i updated to react-native 0.53 and the app runs extremly slow when using react-native-debugger. Using the default browser tab has normal performance

**Comment by andrewkslv** [comment]
I've noticed to same on my side.

**Comment by jhen0409** [comment]
Could you try to close React DevTools and see the performance? I think it should be related, otherwise it shouldn't much difference with the browser.

**Comment by andrewkslv** [comment]
@jhen0409 yeah. I did it and I saw a performance boost. Another thing that I noticed with the debugger if I'm short on CPU/memory app interface becomes unresponsive especially react-navigation.

**Comment by TheDutchCoder** [comment]
For my the ReactNativeDebuggerHelper takes up around 90% of my CPU continually.

It might be related to Hot Module Reloading (that's where it seems to start struggling).

**Comment by pinguinjkeke** [comment]
Same here. The RNDebuggerHelper process memory goes to 2Gb and higher until I'm closing and opening it again

**Comment by joey3060** [comment]
+1 ......
![image](https://user-images.githubusercontent.com/5051756/40266283-d48928dc-5b7a-11e8-9e77-a415cb6630c9.png)

**Comment by Tomas-Sereikis** [comment]
Facing the same issue.

**Comment by Geczy** [comment]
Same issue

**Comment by macrozone** [comment]
it can have different reasons:

- the debugger is slow if you have enabled react-dev-tools (the react tree inspector). Disable that, if you don't use it. This was probably my initial problem here
- your js code runs in the debugger. A slow debugger might indicate a performance problem in the app
- dispatching too many redux-actions can slow down the debugger
- also with redux, there is a more subtle problem: be careful that you not accidentially send large objects as payloads. In particular events from `onPress` or similar. This can happen, if you use `redux-actions` which per default uses the first argument as payload. The redux-devtools can choke on these payloads
